src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,437 @@
+    inner_has_m = (wkb_type_inner & WKB_M) != 0
+    
+    parse_wkb_polygon(io, byte_order_inner, inner_has_z, inner_has_m, crs_type)
+  end
+  
+  # Create Multi geometry - simplified approach
+  return Multi(polygons)
+end
+
+function parse_gpb(blob::Vector{UInt8}, crs_type)
+  header, offset = parse_gpb_header(blob)
+  geometry = parse_wkb_geometry(blob, offset, crs_type)
+  return geometry, header.srs_id
+end
+
+function get_crs_from_srid(db::SQLite.DB, srid::Int32)
+  # Handle special cases
+  if srid == 0
+    return nothing
+  elseif srid == -1
+    return Cartesian{NoDatum}
+  end
+  
+  # Query the database for the organization and organization_coordsys_id
+  query = """
+    SELECT organization, organization_coordsys_id
+    FROM gpkg_spatial_ref_sys
+    WHERE srs_id = ?
+  """
+  
+  srs_result = DBInterface.execute(db, query, [srid])
+  result = []
+  for row in srs_result
+    push!(result, (org=row.organization, coord_id=row.organization_coordsys_id))
+  end
+  
+  if isempty(result)
+    # Default for unknown SRID values
+    @warn "Unknown SRID $srid, defaulting to WGS84"
+    return LatLon{WGS84Latest}
+  end
+  
+  org_info = result[1]
+  
+  # Handle EPSG organization
+  if uppercase(org_info.org) == "EPSG"
+    # Use CoordRefSystems.get with EPSG code
+    epsg_code = org_info.coord_id
+    crs_type = CoordRefSystems.get(EPSG{epsg_code})
+    return crs_type
+  elseif uppercase(org_info.org) == "ESRI"
+    # Handle ESRI codes if needed
+    @warn "ESRI CRS organization detected for SRID $srid, using fallback to WGS84"
+    return LatLon{WGS84Latest}
+  elseif uppercase(org_info.org) == "NONE"
+    # Handle the special "NONE" organization cases
+    if srid == 0
+      return nothing
+    elseif srid == -1
+      return Cartesian{NoDatum}
+    else
+      @warn "Unknown 'NONE' organization SRID $srid, defaulting to WGS84"
+      return LatLon{WGS84Latest}
+    end
+  else
+    # Unknown organization, default to WGS84
+    @warn "Unknown CRS organization '$(org_info.org)' for SRID $srid, defaulting to WGS84"
+    return LatLon{WGS84Latest}
+  end
+end
+
+function gpkgreadattribs(db::SQLite.DB, table_name::String, geom_column::String)
+  # Get all data from the table  
+  data_query = "SELECT * FROM \"$table_name\""
+  data_result = DBInterface.execute(db, data_query)
+  
+  table_data = []
+  for row in data_result
+    # Skip rows with missing geometry
+    blob = getproperty(row, Symbol(geom_column))
+    if !ismissing(blob) && !isnothing(blob)
+      # Create attribute row without geometry column
+      row_dict = Dict()
+      for name in propertynames(row)
+        if name != Symbol(geom_column)
+          row_dict[name] = getproperty(row, name)
+        end
+      end
+      push!(table_data, (; row_dict...))
+    end
+  end
+  
+  # Handle case with no attribute columns (only geometry)
+  if !isempty(table_data) && isempty(first(table_data))
+    return nothing
+  else
+    return Tables.rowtable(table_data)
```

Comment:
Notice that if `table_data` is constructed as a vector of namedtuple, it already implements the Tables.jl interface. A namedtuple of vectors is also a Tables.jl table. Try to construct one of these representations to avoid having to call `Tables.rowtable` and allocating an extra amount of memory.